gcloud/datastore/batch.py

```diff
@@ -193,6 +193,22 @@ def add_auto_id_entity(self, entity):
 
         self._auto_id_entities.append(entity)
 
+    def _match_dataset_id(self, other_id):
```

Comment:
See [`regression.datastore._compare_dataset_ids`][1] for a more strict / preferred method.

I have some tests for this in
https://github.com/dhermes/gcloud-python/tree/key-comparison

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/be8f5fb2ed597060ea1e41c1ce830a3d094b2639/regression/datastore.py#L357